app/src/components/ChatBot/chat-bot.tsx

```diff
@@ -58,20 +66,251 @@ export default function ChatBot({
   t: TFunction;
   inventoryId: string;
 }) {
-  const {
-    messages,
-    input,
-    handleInputChange,
-    handleSubmit,
-    isLoading,
-    append,
-    reload,
-  } = useChat({
-    api: `/api/v0/chat/${inventoryId}`,
-    initialMessages: [
-      { id: "-1", content: t("initial-message"), role: "assistant" },
-    ],
-  });
+  const [threadId, setThreadId] = useState("");
+  const [userInput, setUserInput] = useState("");
+  const [messages, setMessages] = useState<Message[]>([]);
+  const [inputDisabled, setInputDisabled] = useState(false);
+  const [createThreadId] = useCreateThreadIdMutation();
+  const [getAllDataSources] = api.useLazyGetAllDataSourcesQuery();
+  const [getUserInventories] = api.useLazyGetUserInventoriesQuery();
+  const [getInventory] = api.useLazyGetInventoryQuery();
+  const toast = useToast();
+
+  const handleError = (error: any, errorMessage: string) => {
+    console.error(errorMessage, error);
+    // Display error to user (you can use a toast notification or a modal)
+    // Example using Chakra UI's toast:
+    toast({
+      title: "An error occurred",
+      description: errorMessage,
+      status: "error",
+      duration: 5000,
+      isClosable: true,
+    });
+  };
+
+  // Creating the thread id for the given inventory on initial render
+  useEffect(() => {
+    // Function to create the threadId with initial message
+    const initializeThread = async () => {
+      try {
+        const result = await createThreadId({
+          inventoryId: inventoryId,
+          content: t("initial-message"),
+        }).unwrap();
+        setThreadId(result);
+      } catch (error) {
+        handleError(
+          error,
+          "Failed to initialize chat. Please refresh the page.",
+        );
+      }
+    };
+    if (!threadId) {
+      initializeThread();
+    }
+  }, []); // Empty dependency array means this effect runs only once
+
+  // Automatically scroll to bottom of chat
+  const messagesEndRef = useRef<HTMLDivElement | null>(null);
+  const scrollToBottom = () => {
+    messagesEndRef.current?.scrollIntoView({ behavior: "smooth" });
+  };
+  useEffect(() => {
+    scrollToBottom();
+  }, [messages]);
+
+  // TODO: Convert to Redux
+  const sendMessage = async (text: string) => {
+    try {
+      const response = await fetch(`/api/v0/assistants/threads/messages`, {
+        method: "POST",
+        body: JSON.stringify({
+          threadId: threadId,
+          content: text,
+        }),
+      });
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
```

Comment:
we might want to log this error so we can track it and fix it if needed

Reply:
```suggestion
        const data = await response.text();
        console.log("HTTP response text", data);
        throw new Error(`HTTP error! status: ${response.status}`);
```

Reply:
I added this code as console.error. And removed the console.error in handleError (which was only logging the error code but not the error text)